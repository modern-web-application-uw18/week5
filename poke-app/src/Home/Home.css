footer {
  display: flex;
  flex-flow: row;
  justify-content: space-evenly;
}

footer div {
  cursor: pointer;
}

.btn {
  padding: 5px;
  border: 1px solid #181818;
  border-radius: 10px;
}

.pokemon {
  padding: 5px;
  border: 1px solid rgb(244, 244, 244);
  margin: 5px;
  text-decoration: none;
  border-radius: 10px;
  cursor: pointer;
}

.pokemon:hover {
  border: 1px solid #EE343B;
}

.name {
  text-transform: capitalize;
}

.description {
  font-size: 12px;
}

.nextPage {
  cursor: pointer;
}

footer {
  margin: 5px;
  padding: 5px;
}

header ul {
  display: flex;
  flex-flow: row;
  justify-content: space-evenly;
  list-style-type: none;
}
