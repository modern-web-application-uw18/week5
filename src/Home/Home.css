.home .page-content {
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  flex-wrap: wrap;
}

.home .pagination {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.home .pagination button {
  font-size: 1.2rem;
  margin: 0 1rem;
  color: #585858;
  cursor: pointer;
}
