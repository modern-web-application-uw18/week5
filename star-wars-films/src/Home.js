import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link, ListItemLink} from 'react-router-dom';
import PersonDetail from './PersonDetail';
import Characters from './characters.json';
import Avatar, { AvatarItem } from '@atlaskit/avatar';
import { Container, Row, Col } from 'react-grid-system';

class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
          people: []
        };
      }

      componentDidMount() {
        fetch('https://swapi.co/api/people')
        .then(response => response.json())
        .then(data => {
          this.setState((prevState, props) => {
            return {
              people: data.results,
              characters: this.props.Characters,
            };
          });
        })
        .catch(error => console.log(error));
      }

    render() {
        const charImage = Characters.characters;
        console.log(charImage);
      return (
        <Router>
        <div>
          <h1>Star Wars Characters</h1>
          <h5>Click on any of the following for a bio</h5>
          <br/>
          <Container>
            <Row align="center">
              
                {charImage.character.map(
                  character =>
                  <Col ><AvatarItem avatar={<Avatar src={character.thumbnail} size="large" appearance="square"/>}/></Col>
                )}
              
            </Row>
            <Row align="center" >
              
                {this.state.people.map(
                people => 
                <Col ><Link to={`/people/${people.name}`}><h5>{people.name}</h5></Link></Col>
                )}
              
            </Row>
            </Container>
        </div>
        </Router>
      )
    }
  }

export default Home;