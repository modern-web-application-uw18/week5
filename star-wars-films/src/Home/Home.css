.home.page-content {
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  flex-wrap: wrap;
  padding: 0 1.5rem;
}
