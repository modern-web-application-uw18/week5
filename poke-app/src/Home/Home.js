import React, { Component } from 'react';

import './Home.css';

import {
  BrowserRouter as Router, Route, Link
} from 'react-router-dom';

export default class Home extends Component {

    render() {
      /**
      * Loop through the props
      */
      const pokemonList = this.props.pokeList.map((pokemon, index) => {
        const { name, url, sprite } = pokemon;
        const arr = url.split('/').filter(word => word !== '');
        const id = arr[arr.length - 1];

        return (
          <Link to={`/description/${id}`} key={name} >
            <div onClick={this.props.fetchURL(url)} className="pokemon" title={name}>
              <div className="image">
                <img src={sprite} alt={name} />
              </div>
              <div className="name">{name}</div>
            </div>
          </Link>
        )
      });

        return (
          <main>
            {pokemonList}
          </main>
        );
    }
}

// Home.defaultProps = {
//   pokeList: [],
//   pokemon: {
//     name: 'charizard',
//     url: '',
//     sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png'
//   }
// }
