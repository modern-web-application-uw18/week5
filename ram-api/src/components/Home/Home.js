import React, { Component } from 'react';
import Card from './Card/Card';
import './Home.css';
import PropTypes from 'prop-types';
import Paginator from './Paginator/Paginator';


class Home extends Component {

    state = {
        idxFrom: 0,
        idxTo: 19,
        first: 1,
        second: 2,
        third: 3,
        fourth: 4,
        fifth: 5,
        searchBoxValue: ''
    }


    componentDidUpdate(prevProps, prevState) {

        //if user types into the search area, go back to the first page of the paginator
        if (prevProps.searchBoxValue.toLowerCase() !== this.props.searchBoxValue.toLowerCase()) {
            this.setState(prevState => {
                return {
                    idxFrom: 0,
                    idxTo: 19,
                    first: 1,
                    second: 2,
                    third: 3,
                    fourth: 4,
                    fifth: 5
                }
            })
        }
    }


    paginatorHandler = (e, buttonID, innerValue) => {

        switch (buttonID) {
            case "inner":
                this.setState(prevState => {
                    let _idxFrom, _idxTo;

                    _idxFrom = innerValue === 1 ? innerValue * 19 : innerValue * 19 + 1;
                    _idxTo = _idxFrom + 19;

                    return {
                        idxFrom: _idxFrom,
                        idxTo: _idxTo
                    }
                })
                break;

            case "prev":

                this.setState(prevState => {
                    let _first, _second, _third, _fourth, _fifth;
                    _first = (prevState.first != null) && (prevState.first > 1) ? --prevState.first : 1;
                    _second = (prevState.second != null) && (prevState.second > 2) ? --prevState.second : 2;
                    _third = (prevState.third != null) && (prevState.third > 3) ? --prevState.third : 3;
                    _fourth = (prevState.fourth != null) && (prevState.fourth > 4) ? --prevState.fourth : 4;
                    _fifth = (prevState.fifth != null) && (prevState.fifth > 5) ? --prevState.fifth : 5;
                    return {
                        first: _first,
                        second: _second,
                        third: _third,
                        fourth: _fourth,
                        fifth: _fifth,
                        idxFrom: _first === 1 ? 0 : _first * 19 + 1,
                        idxTo: _first === 1 ? 19 : _first * 19 + 19
                    }
                })
                break;


            case "next":
                this.setState(prevState => {
                    let _first, _second, _third, _fourth, _fifth;
                    _first = (prevState.first != null) && (prevState.first < this.props.pageCount - 4) ? ++prevState.first : this.props.pageCount - 4;
                    _second = (prevState.second != null) && (prevState.second < this.props.pageCount - 3) ? ++prevState.second : this.props.pageCount - 3;
                    _third = (prevState.third != null) && (prevState.third < this.props.pageCount - 2) ? ++prevState.third : this.props.pageCount - 2;
                    _fourth = (prevState.fourth != null) && (prevState.fourth < this.props.pageCount - 1) ? ++prevState.fourth : this.props.pageCount - 1;
                    _fifth = (prevState.fifth != null) && (prevState.fifth < this.props.pageCount) ? ++prevState.fifth : this.props.pageCount;
                    return {
                        first: _first,
                        second: _second,
                        third: _third,
                        fourth: _fourth,
                        fifth: _fifth,
                        idxFrom: _first === 1 ? 0 : _first * 19 + 1,
                        idxTo: _first === 1 ? 19 : _first * 19 + 19

                    }
                })
                break;




            default:
                this.setState(prevState => {
                    return {
                        idxFrom: 0,
                        idxTo: 19,
                        first: 1,
                        second: 2,
                        third: 3,
                        fourth: 4,
                        fifth: 5
                    }
                })
        }

    }


    render() {



        let filteredCards = this.props.characters.filter(character => {
            return character.name ? character.name.toLowerCase().indexOf(this.props.searchBoxValue.toLowerCase()) !== -1 : null;
        });


        return (
            <div style={{ textAlign: "center" }}>

                {/* //show the paginator only when there is no filtering (searchbox value) in place */}
                {this.props.searchBoxValue.length === 0 ?
                    <Paginator
                        pageCount={this.props.pageCount}
                        characterCount={this.props.characterCount}
                        paginatorHandler={this.paginatorHandler}
                        first={this.state.first}
                        second={this.state.second}
                        third={this.state.third}
                        fourth={this.state.fourth}
                        fifth={this.state.fifth}
                    />
                    :
                    null


                }


                <div className="Home">
                    {
                        //show the limited amount of cards only when there is no filtering (searchbox value) in place
                        //otherwise show everything
                        this.props.searchBoxValue.length === 0 ?
                            filteredCards.map((character, idx) => {
                                return idx >= this.state.idxFrom && idx <= this.state.idxTo ? <Card key={character.id} character={character} /> : null;
                            })
                            :
                            filteredCards.map((character, idx) => {
                                return <Card key={character.id} character={character} />;
                            })
                    }
                </div>
            </div>
        )

    }

}


Home.propTypes = {
    characters: PropTypes.array.isRequired,
    pageCount: PropTypes.number,
    characterCount: PropTypes.number,
    searchBoxValue: PropTypes.string
}


export default Home;
