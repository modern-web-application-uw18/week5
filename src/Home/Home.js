import React from 'react';
import axios from 'axios';
import PersonCard from '../PersonCard/PersonCard';
import './Home.css';

import PropTypes from 'prop-types';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      people: [],
      page: this.props.page
    }
  }

  componentDidMount = async () => {
    let url = `https://swapi.dev/api/people/?page=${this.state.page}`;
    let res = await axios.get(url);
    this.setState({
        people: res.data.results
    });
  }

  next = () => {
    this.setState((prevState) => {
      return { page: prevState.page + 1 };
    }, this.componentDidMount);
  }

  previous = () => {
    this.setState((prevState) => {
      return { page: prevState.page - 1 };
    }, this.componentDidMount);
  }

  getPrevButton() {
    return (
      <button onClick={this.previous}>Previous</button>
    );
  }

  getNextButton() {
    return (
      <button onClick={this.next}>Next</button>
    );
  }

  render () {
    const people = this.state.people;
    return (
      <div className="home">
        <section className="page-content">
          {people.map((people, idx) => {
            return <PersonCard key={idx} people={people} />;
          })}
        </section>
        <div className="pagination">
          {this.state.page > 1 ? this.getPrevButton() : null}
          {this.state.page < 9 ? this.getNextButton() : null}
        </div>
      </div>
    )
  }
}

export default Home;


Home.propTypes = {
  page: PropTypes.number.isRequired
};
