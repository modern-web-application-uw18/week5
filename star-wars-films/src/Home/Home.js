import React, { Component } from 'react';
import PersonCard from '../PersonCard/PersonCard';
import {
  Link
} from "react-router-dom";
import './Home.css';

class home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      people: [],
      page: 1
    }
  }

  componentDidMount() {
    fetch(`https://swapi.co/api/people/?page=${this.state.page}`)
      .then(response => response.json())
      .then(data => {
        if (data.results.length > 0) {
        this.setState((prevState, props) => {
          return {
            people: data.results
          };
        });
      } else {
        this.setState((prevState, props) => {
          return {
            people: []
          };
        });
      }
      })
      .catch(error => console.log(error));
  };

  next = () => {
    this.setState((prevState) => {
      return { page: prevState.page + 1 };
    }, this.componentDidMount);
  }

  previous = () => {
    this.setState((prevState) => {
      return { page: prevState.page - 1 };
    }, this.componentDidMount);
  }

  getPrevButton() {
    return (
      <button onClick={this.previous}>Previous</button>
    );
  }

  getNextButton() {
    return (
      <button onClick={this.next}>Next</button>
    );
  }

  render() {
    const people = this.state.people;
    return (
      <div className="home">
        <section className="page-content">
          {people.map((people, idx) => {
            return <PersonCard key={idx} people={people} />;
          })}
        </section>
        <div className="pagination">
          {this.state.page > 1 ? this.getPrevButton() : null}
          {this.state.page < 9 ? this.getNextButton() : null}
        </div>
      </div>
    )
  }
}

export default home;
