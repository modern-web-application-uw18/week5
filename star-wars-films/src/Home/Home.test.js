import React from 'react';
import Home from './Home';
import ReactTestRenderer from 'react-test-renderer';
import {
  MemoryRouter
} from "react-router-dom";

describe('Home component', () => {
  it('should render', () => {
    const component = ReactTestRenderer.create(
      <MemoryRouter>
        <Home page={1}/>
      </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
