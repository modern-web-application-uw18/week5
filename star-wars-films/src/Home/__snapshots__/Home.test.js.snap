// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render 1`] = `
<div
  className="home"
>
  <section
    className="page-content"
  />
  <div
    className="pagination"
  >
    <button
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
